shell/platform/common/cpp/client_wrapper/plugin_registrar_unittests.cc

```diff
@@ -34,8 +35,34 @@ class TestApi : public testing::StubFlutterApi {
 
   const uint8_t* last_data_sent() { return last_data_sent_; }
 
+  int64_t RegisterExternalTexture(
+    FlutterTexutreCallback texture_callback,
+    void* user_data) override {
+      last_texture_id_ = 1;
+      return last_texture_id_;
+  }
+
+  bool UnregisterExternalTexture(int64_t texture_id) override {
+    if(texture_id == 2) {
+      last_texture_id_ = -2;
+      return true;
+    }
+    return false;
+  }
+
+  bool TextureFrameAvailable(int64_t texture_id) override {
+     if(texture_id == 1) {
+      last_texture_id_ = 2;
```

Comment:
I'm confused by all the magic values here. Please add comments in the stub explaining what this is doing.

Although, what it looks like from reading the test and this code together is that these values are carefully chose to do exactly what that one test expects, which is not ideal since, for instance, it would be impossible to write a second test that did something different. It's also extremely confusing to me that calling TextureFrameAvailable changes the ID since that isn't at all what the actual implementation would do, and results in the test having code that appears to be actively wrong, asserting that unregistering an ID *that was never registered* works.

It would be a lot easier to understand this if the stub were slightly more generic. For instance:
- Make an ivar with a `map` of registered texture IDs, which register/unregister would update as expected and return success or failure accordingly (e.g., failing registration if already registered, and unregistration if not registered)
- Make a test method to query whether a texture is registered (i.e., in the map) for use in verifying that the calls were made as expected.
- Make TextureFrameAvailable increment the value in the map for that texture ID, and similarly add a test validation query.

With that structure, the stub will make sense without going and reading the test, the test would read in a clear and obvious way instead of having a series of calls that look wrong, and other tests could be written without complex additional logic here.